common/logging.go

```diff
@@ -82,56 +86,143 @@ func (l ConsoleLogger) IsLogLevel(level LogLevel) bool {
 	}
 }
 
+// Trace logs 'Trace' message.
 func (l ConsoleLogger) Trace(format string, args ...interface{}) {
 	if l.LogLevel >= LogLevelTrace {
 		prefix := "[TRACE] "
 		l.output(os.Stdout, prefix, format, args...)
 	}
 }
 
+// output writes `format`, `args` log message prefixed by the source file name, line and `prefix`
+//noinspection GoUnhandledErrorResult
+func (l ConsoleLogger) output(f io.Writer, prefix string, format string, args ...interface{}) {
+	output(f, prefix, format, args...)
+}
+
 var Log Logger = DummyLogger{}
 
+// SetLogger sets 'logger' to be used by the unidoc unipdf library.
 func SetLogger(logger Logger) {
 	Log = logger
 }
 
+// WriterLogger is the logger that writes data to the Output writer
+type WriterLogger struct {
+	LogLevel LogLevel
+	Output   io.Writer
+}
+
+// NewWriterLogger creates new writer logger.
+func NewWriterLogger(logLevel LogLevel, writer io.Writer) *WriterLogger {
+	logger := WriterLogger{
+		Output:   writer,
+		LogLevel: logLevel,
+	}
+	return &logger
+}
+
+// IsLogLevel returns true if log level is greater or equal than `level`.
+// Can be used to avoid resource intensive calls to loggers.
+func (l WriterLogger) IsLogLevel(level LogLevel) bool {
+	return l.LogLevel >= level
+}
+
+// Error logs 'Error' message.
+func (l WriterLogger) Error(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelError {
+		prefix := "[ERROR] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
+// Warning logs 'warning' message.
+func (l WriterLogger) Warning(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelWarning {
+		prefix := "[WARNING] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
+// Notice logs 'Notice' message.
+func (l WriterLogger) Notice(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelNotice {
+		prefix := "[NOTICE] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
+// Info logs 'Info' message.
+func (l WriterLogger) Info(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelInfo {
+		prefix := "[INFO] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
+// Debug logs 'Debug' message.
+func (l WriterLogger) Debug(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelDebug {
+		prefix := "[DEBUG] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
+// Trace logs 'Trace' message.
+func (l WriterLogger) Trace(format string, args ...interface{}) {
+	if l.LogLevel >= LogLevelTrace {
+		prefix := "[TRACE] "
+		l.output(l.Output, prefix, format, args...)
+	}
+}
+
 // output writes `format`, `args` log message prefixed by the source file name, line and `prefix`
-func (l ConsoleLogger) output(f *os.File, prefix string, format string, args ...interface{}) {
+//noinspection GoUnhandledErrorResult
+func (l WriterLogger) output(f io.Writer, prefix string, format string, args ...interface{}) {
+	output(f, prefix, format, args)
+}
+
+func output(f io.Writer, prefix string, format string, args ...interface{}) {
```

Comment:
I think the name of this function should be changed to something a bit more suggestive. As part of a struct, it did provide some context. Maybe `logToWriter` or `logOutput` or something like that.